aml/src/stream.rs

```diff
@@ -0,0 +1,339 @@
+use core::{convert::TryInto, mem};
+
+/// A container for the buffer that provides debugging info
+#[derive(Copy, Clone)]
+pub struct AmlStream<'a> {
+    /// The address of the original data, used for debugging only
+    stream_start: usize,
```

Comment:
Hm, I always imagined storing an offset from the beginning of the stream, rather than the raw pointer of the memory itself. I don't think that would have less debugging value - what do you think?